src/components/InputRange.vue

```diff
@@ -28,19 +28,32 @@ const Input = require('gui/inputs/input');
 export default {
   mixins: [Input],
   data() {
-    const options = this.state.input.options.values[0];
-    const min = 1*options.min;
-    const max = 1*options.max;
-    const step = 1*options.Step;
+    const {min, max, Step} = this.state.input.options.values[0];
     return {
-      max,
-      min,
-      step: step
+      min: 1*min,
+      max: 1*max,
+      step: 1*Step
     }
   },
   methods: {
     checkValue() {
-      const valid = this.state.validate.required || !_.isEmpty(_.trim(this.state.value)) ? this.service.getValidator().validate(this.state.value) : true;
+      // set initial valid to true
+      let valid = true;
+      // check if value of input is empty
+      const isEmpty = _.isEmpty(_.trim(this.state.value));
+      // in case of required input check if is not empty and check validity from validator
+      if (this.state.validate.required) {
+        valid =  !isEmpty && this.service.getValidator().validate(this.state.value);
+      } else {
+        //in case not required check if value is empty
+        if (isEmpty) {
+          //set default value
+          this.state.value = this.state.input.options.values[0].default;
+        } else {
+          // check validity from validator
+          valid = this.service.getValidator().validate(this.state.value);
+        }
+      }
```

Comment:
Not a big fan of this kind of solution:

1. if you're building a validator, it's always best to start with a false assumption

```js
// set initial valid to true
let valid = true;
```

2. if possible it is best to avoid nested else-if conditions, as they make the code more difficult to read and debug. Whenever possible consider writing simpler conditionals that remains at the same "indentation level" (it's also good for your brain..):

```suggestion
    checkValue() {

      // set initial valid to true
      let valid = true;

      // check if value of input is empty
      const isEmpty = _.isEmpty(_.trim(this.state.value));

      // required input, check if is not empty and check validity from validator
      if (this.state.validate.required) {
        valid = !isEmpty && this.service.getValidator().validate(this.state.value);
      }

      // not required input, check if value is empty and set default value
      else if (isEmpty) {
        this.state.value = this.state.input.options.values[0].default;
      }

       // check validity from validator
      else {
        valid = this.service.getValidator().validate(this.state.value);
      }

      this.state.validate.valid = valid;
      this.change();

    }
```